### PR TITLE
compound.finance message support

### PR DESCRIPTION
**Pull request by keepkeyjon**
Docs: https://compound.finance/developers

- [ ] Tests
- [x] Comptroller enter market
- [x] Comptroller exit market

### CODE REVIEW

File: include/keepkey/firmware/ethereum_contracts/compound.h [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef KEEPKEY_FIRMWARE_ETHEREUMCONTRACTS_COMPUND_H
+#define KEEPKEY_FIRMWARE_ETHEREUMCONTRACTS_COMPUND_H
```
Comment (keepkeyjon) [inline include/keepkey/firmware/ethereum_contracts/compound.h]: s/COMPUND/COMPOUND/g

File: lib/firmware/ethereum_contracts/compound.c [hunk 2]
```diff
@@ -0,0 +1,526 @@
...
+     { 1, "\x6c\x8c\x6b\x02\xe7\xb2\xbe\x14\xd4\xfa\x60\x22\xdf\xd6\xd7\x59\x21\xd9\x0e\x4e", "cBAT",   8}},
+    {{ 1, "\x89\xd2\x4a\x6b\x4c\xcb\x1b\x6f\xaa\x26\x25\xfe\x56\x2b\xdd\x9a\x23\x26\x03\x59", "DAI",   18},
+     { 1, "\xf5\xdc\xe5\x72\x82\xa5\x84\xd2\x74\x6f\xaf\x15\x93\xd3\x12\x1f\xca\xc4\x44\xdc", "cDAI",   8}},
+    {{ 1, "\x19\x85\x36\x5e\x9f\x78\x35\x9a\x9B\x6A\xD7\x60\xe3\x24\x12\xf4\xa4\x45\xE8\x62", "REP",   18},
+     { 1, "\x15\x80\x79\xee\x67\xfc\xe2\xf5\x84\x72\xa9\x65\x84\xa7\x3c\x7a\xb9\xac\x95\xc1", "cREP",   8}},
+    {{ 1, "\xa0\xb8\x69\x91\xc6\x21\x8b\x36\xc1\xd1\x9d\x4a\x2e\x9e\xb0\xce\x36\x06\xeb\x48", "USDC",   6},
+     { 1, "\x39\xaa\x39\xc0\x21\xdf\xba\xe8\xfa\xc5\x45\x93\x66\x93\xac\x91\x7d\x5e\x75\x63", "cUSDC",  6}},
+    {{ 1, "\xe4\x1d\x24\x89\x57\x1d\x32\x21\x89\x24\x6d\xaf\xa5\xeb\xde\x1f\x46\x99\xf4\x98", "ZRX",   18},
+     { 1, "\xb3\x31\x9f\x5d\x18\xbc\x0d\x84\xdd\x1b\x48\x25\xdc\xde\x5d\x5f\x72\x66\xd4\x07", "cZRX",   8}},
+};
```
Comment (keepkeyjon) [inline lib/firmware/ethereum_contracts/compound.c]: Should add Kovan & Rinkeby cETH contracts.

File: lib/firmware/ethereum_contracts/compound.c [hunk 3]
```diff
@@ -0,0 +1,526 @@
...
+{
+    return msg->data_initial_chunk.size == 4 + count * 32;
+}
+
+static inline const uint8_t *getParam(const EthereumSignTx *msg, size_t idx)
+{
+    return msg->data_initial_chunk.bytes + 4 + idx * 32;
+}
+
+static bool isMethod(const EthereumSignTx *msg, const char *hash,
```
Comment (keepkeyjon) [inline lib/firmware/ethereum_contracts/compound.c]: Copied these from `makerdao.c` to get it working, but they should really live in some common spot instead.

File: lib/firmware/ethereum_contracts/compound.c (hunk 2 above)
Comment (keepkeyjon) [reply lib/firmware/ethereum_contracts/compound.c]: Also need to make sure they're in the `TokenType` table, so we get `Approve` & `Transfer` support for them too.

File: lib/firmware/ethereum_contracts/compound.c [hunk 4]
```diff
@@ -0,0 +1,526 @@
...
+     { 1, "\x15\x80\x79\xee\x67\xfc\xe2\xf5\x84\x72\xa9\x65\x84\xa7\x3c\x7a\xb9\xac\x95\xc1", "cREP",   8}},
+    {{ 1, "\xa0\xb8\x69\x91\xc6\x21\x8b\x36\xc1\xd1\x9d\x4a\x2e\x9e\xb0\xce\x36\x06\xeb\x48", "USDC",   6},
+     { 1, "\x39\xaa\x39\xc0\x21\xdf\xba\xe8\xfa\xc5\x45\x93\x66\x93\xac\x91\x7d\x5e\x75\x63", "cUSDC",  6}},
+    {{ 1, "\xe4\x1d\x24\x89\x57\x1d\x32\x21\x89\x24\x6d\xaf\xa5\xeb\xde\x1f\x46\x99\xf4\x98", "ZRX",   18},
+     { 1, "\xb3\x31\x9f\x5d\x18\xbc\x0d\x84\xdd\x1b\x48\x25\xdc\xde\x5d\x5f\x72\x66\xd4\x07", "cZRX",   8}},
+};
+
+static CompoundContract *contractByAddress(const uint8_t *address)
+{
+    for (int i = 0; i < sizeof(contracts)/sizeof(contracts[0]); i++) {
```
Comment (keepkeyjon) [inline lib/firmware/ethereum_contracts/compound.c]: signed/unsigned compare. needs to be `size_t`

**Review by Aggrord09 — APPROVED** [review]

**Pull request closed without merging**